### ISSUE TITLE
Change tracking on components holding handles to assets don't trigger when the asset handle transitions from the loading to the loaded state

### ISSUE THREAD

**Issue by Neurrone** [bevyengine/bevy#2688]
## Bevy version

0.5

## Operating system & version

Windows 10

## What you did

Have a component hold a `Handle<T>` of some asset. In my case, I'm working on integration with the Synthizer audio library and have Sound components which take in a handle to the `Buffer` of audio to use. 

When the asset finishes loading, then the behaviour I wanted is for the sound to e.g start playing

Hence, I have a query on `Changed<Sound>` to know when Sound components have been changed and logic should be rerun.

## What you expected to happen

Change tracking work for resource handles on components when they transition from loading to loaded states.

## What actually happened

Change tracking doesn't work in this case.

## Additional information

Any additional information you would like to add such as screenshots, logs, etc.

**Comment by mockersf** [comment]
Yup this is a known issue...
In Bevy there's one place where this happens, as a `ColorMaterial` has an `Handle<Texture>`.

Workaround is here:https://github.com/bevyengine/bevy/blob/73f4a9d18f609e52d28e2963635560e0b78a2f9f/crates/bevy_sprite/src/color_material.rs#L60-L63

A more general solution would be nice 👍

**Comment by cart** [comment]
Yeah I'm open to alternatives to the current "asset change events", but I don't want to overload the semantics of `Handle<T>` changing. That should only indicate that the "handle component value" has changed, not the asset it points to.

**Comment by alice-i-cecile** [comment]
> That should only indicate that the "handle component value" has changed, not the asset it points to.

Hmm. I kind of like the idea of a specialized `AssetChanged` query filter (and corresponding `.is_asset_changed` method) in order to allow users to opt-in to this information. That should be fairly easy to implement.

There are two possible designs here:
1. Returns true if the underlying asset data changed, returns false if only the address stored in the handle changed.
2. Returns true if the underlying asset data changed or the address stored in the handle changed.

The first one is properly disjoint from `Changed`, but I can't imagine this behavior would be commonly wanted. 

The second design would push us towards adding a `Not` query filter again, but that's easy. I think this is my preferred approach.

**Comment by Neurrone** [comment]
@mockersf Thanks for that link. There's one part which I don't understand, why the use of the ManualEventReader vs a normal EventReader for `material_events_reader`? What's the difference?
